src/google/protobuf/compiler/code_generator.h

```diff
@@ -176,6 +176,11 @@ class PROTOC_EXPORT GeneratorContext {
   // this GeneratorContext.
   virtual void GetCompilerVersion(Version* version) const;
 
+  virtual void SetCurrentFile(const FileDescriptor *file);
```

Comment:
I don't personally like this, but I didn't see a simpler way to bridge the gap between a FileDescriptor and the paths of generated files derived from it. Hopefully someone has a suggestion for this that's a bit more elegant.